accepted/2025/sdk-selection-improvements.md

```diff
@@ -0,0 +1,269 @@
+
+An attempt to improve SDK selection was made by modifying the SDK resolution path via [dotnet/runtime PR #118092](https://github.com/dotnet/runtime/pull/118092). However, this change was [reverted in PR #120472](https://github.com/dotnet/runtime/pull/120472) due to a critical issue: when uninstalling a newer .NET version, the PATH entry to `dotnet.exe` would be incorrectly removed even when older .NET versions remained installed. For example, if .NET 9.0 was installed first, then .NET 10 RC1, uninstalling .NET 10 RC1 would remove the PATH to `dotnet.exe` entirely, breaking the remaining .NET 9.0 installation.
+
+This revert highlighted the complexity of PATH management in multi-version .NET environments and left gaps in flexibility and user experience. The goal now is to provide a more robust and configurable approach for SDK selection across platforms that avoids these PATH management pitfalls.
+
+## Related Documentation
+
+*   [.NET Environment Variables](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables) - Official documentation for .NET environment variables
+*   [global.json Overview](https://learn.microsoft.com/en-us/dotnet/core/tools/global-json) - Documentation for global.json configuration
+*   [.NET Installation Guide](https://learn.microsoft.com/en-us/dotnet/core/install/) - Official installation documentation
+*   [SDK Resolution in .NET](https://github.com/dotnet/designs/blob/main/accepted/2019/sdk-version-selection.md) - Design document for SDK version selection
+
+***
+
+## Non-Goals
+
+*   **SDK Location Reporting API:** This proposal does not include an API or mechanism for the .NET host or `dotnetup` to report back the location where the SDK is installed. Such functionality for programmatic SDK discovery will be covered in a separate design effort.
+
+***
+
+## Proposal: DOTNET_ROOT_REDIRECT
+
+Introduce two new environment variables that allow redirecting SDK loading from a specific source location to a target location:
+
+- `DOTNET_ROOT_REDIRECT_SOURCES`: Specifies the source SDK installation path to redirect from (e.g., `C:\Program Files\dotnet`)
+- `DOTNET_ROOT_REDIRECT_TARGET`: Specifies the target SDK installation path to redirect to (e.g., `C:\Users\Daniel\AppData\Local\dotnet`)
+
+### Behavior
+
+*   These environment variables only affect SDK loading behavior when running `dotnet` commands that require SDK resolution.
+*   Direct execution of applications via `dotnet foo.dll` is **not affected** to avoid impacting runtime behavior.
+*   If the host is launched from the source path specified in `DOTNET_ROOT_REDIRECT_SOURCES`, SDK resolution will use the path specified in `DOTNET_ROOT_REDIRECT_TARGET`.
```

Comment:
I dislike CLI tools that change the behavior based on the path they are launched from. It means e.g. copying/linking the tool to some other place or launching it via an alternative path produces different results...